packages/core/src/progress-circle/progress-circle.element.spec.ts

```diff
@@ -155,4 +155,160 @@ describe('progress circle element – ', () => {
+      expect(component.getAttribute('aria-label')).toBe('Now Loading... 49 out of 100 items');
+    });
+  });
+
+  describe('ariaLabel', () => {
+    it('should show indeterminate label if no label is set and there is no value', async () => {
+      await componentIsStable(componentUnset);
+      expect(componentUnset.getAttribute('aria-label')).toBe('Loading');
+    });
+    it('should show loading label and value if no label is set', async () => {
+      await componentIsStable(component);
+      expect(component.getAttribute('aria-label')).toBe('Loading 49%');
+      component.value = 80;
+      await componentIsStable(component);
+      expect(component.getAttribute('aria-label')).toBe('Loading 80%');
+    });
+    it('should not show default label if a custom label is set', async () => {
+      await componentIsStable(componentUnset);
+      expect(componentUnset.getAttribute('aria-label')).toBe('Loading');
+      componentUnset.setAttribute('aria-label', 'ohai');
+      await componentIsStable(componentUnset);
+      expect(componentUnset.getAttribute('aria-label')).toBe('ohai');
+
+      await componentIsStable(component);
+      component.setAttribute('aria-label', 'ohai');
+      await componentIsStable(component);
+      expect(component.getAttribute('aria-label')).toBe('ohai');
+      component.value = 88;
+      await componentIsStable(component);
+      expect(component.getAttribute('aria-label')).toBe('ohai');
+    });
+  });
+
+  describe('i18n', () => {
+    const newLoading = 'ohai';
+    const newLooping = 'kthxbye';
+
+    it('uses i18n strings by default', async () => {
+      await componentIsStable(component);
+      expect(component.getAttribute('aria-label').indexOf(component.i18n.loading) > -1).toBe(
+        true,
+        'has loading i18n string'
+      );
+      await componentIsStable(componentUnset);
+      expect(componentUnset.getAttribute('aria-label').indexOf(componentUnset.i18n.looping) > -1).toBe(
+        true,
+        'has looping i18n string'
+      );
+    });
+    it('updates looping i18n strings as expected', async () => {
+      await componentIsStable(componentUnset);
+      componentUnset.setAttribute('cds-i18n', `{ "loading": "${newLoading}", "looping": "${newLooping}" }`);
+      await componentIsStable(componentUnset);
+      expect(componentUnset.getAttribute('aria-label').indexOf(newLooping) > -1).toBe(true);
+    });
+    it('updates loading i18n strings as expected', async () => {
+      await componentIsStable(component);
+      component.setAttribute('cds-i18n', `{ "loading": "${newLoading}", "looping": "${newLooping}" }`);
+      await componentIsStable(component);
+      expect(component.getAttribute('aria-label').indexOf(newLoading) > -1).toBe(true);
+    });
+  });
+
+  describe('setAriaAttributes', () => {
+    it('should return expected for indeterminate progress', async () => {
+      await componentIsStable(componentUnset);
+      componentUnset.setAriaAttributes();
```

Comment:
Why do we need this call? Initialize aria attributes manually? Why would we want it uninitialized.

note from myself after reading more code:
Obviously it resets/triggers the defaults for the min/max/now attributes. I still don't have a good understanding about the balance between automation and manual control with this approach.

Reply:
the automation here is a convenience. setting them manually at this point is a way to dodge issues with how react supports aria attributes (specifically how it struggles with removing aria attributes). once that support improves, we may be able to set these as properties on the host component and derive their values through getters or as a set default.